### PR TITLE
Update dependencies and AGP version

### PR DESCRIPTION
**Pull request by DrakslerT**
## Summary

Update dependencies and AGP to 9.3.0

**Related issue**: 

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [x] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [x] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: config.gradle [hunk 1]
```diff
@@ -2,8 +2,7 @@ ext {
     buildConfig = [
             "minSdk"    : 23,
             "compileSdk": 36,
-            "targetSdk" : 36,
-            "buildTools": "35.0.0"
```
Comment (DrakslerT) [inline config.gradle:6]: the default `buildTools` is now `36.0.0`

File: gradle.properties [hunk 2]
```diff
@@ -11,15 +11,5 @@ org.gradle.jvmargs=-Xmx2048m
...
-# https://developer.android.com/topic/libraries/support-library/androidx-rn
-android.useAndroidX=true
-# Automatically convert third-party libraries to use AndroidX
-android.enableJetifier=false
 # Kotlin code style for this project: "official" or "obsolete":
-kotlin.code.style=official
-kapt.use.worker.api=true
-android.defaults.buildfeatures.buildconfig=true
-android.nonTransitiveRClass=false
-android.nonFinalResIds=false
```
Comment (DrakslerT) [inline gradle.properties:25]: These are now default values after AGP 9

File: processor/build.gradle [hunk 3]
```diff
@@ -30,13 +30,17 @@ dependencies {
...
+    // mask this.
     implementation project(":annotations")
-    compileOnly project(":core")
-    compileOnly packages.kotlin.serialization_core
-    compileOnly packages.retrofit.core
-    compileOnly project(":retrofit")
+    implementation project(":core")
+    implementation packages.kotlin.serialization_core
+    implementation packages.retrofit.core
+    implementation project(":retrofit")
```
Comment (KCeh) [inline processor/build.gradle:43]: Are we 100% this won't cause build issues on other projects?
Like Duplicate class error

File: sample/build.gradle [hunk 4]
```diff
@@ -1,22 +1,20 @@
 plugins {
     id "com.android.application"
-    id "kotlin-android"
     id "kotlinx-serialization"
-    id "kotlin-kapt"
+    id "com.android.legacy-kapt"
     id "dagger.hilt.android.plugin"
 }
```
[the remark this answers is not in the document]
Comment (KCeh) [reply sample/build.gradle:6]: 👀

File: build.gradle [hunk 5]
```diff
@@ -12,6 +12,7 @@ buildscript {
 
     dependencies {
         classpath packages.gradle
+        classpath packages.legacy_kapt
```
Comment (KCeh) [inline build.gradle:15]: Why is this needed?

File: dependencies.gradle [hunk 6]
```diff
@@ -1,5 +1,5 @@
 ext.versions = [
-        "gradle"       : '8.13.0',
+        "gradle"       : '9.2.1',
```
Comment (KCeh) [inline dependencies.gradle]: This is not the latest AGP: https://developer.android.com/build/releases/agp-9-3-0-release-notes
We need to update to latest for bot to properly dismiss alerts

**Review by KCeh — COMMENTED** [review]
Looks mostly good, 
check out CoPilot comments and my comment about latest AGP version

File: build.gradle (hunk 5 above)
Comment (DrakslerT) [reply build.gradle:15]: Before,`"kotlin-kapt"` was already bundled in `classpath packages.kotlin.plugin`.

`"com.android.legacy-kapt"` is not bundled there, so it needs to be added explicitly through `classpath packages.legacy_kapt`.

If I remove this, the sample module fails to build/sync.

File: sample/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (DrakslerT) [reply sample/build.gradle:6]: https://developer.android.com/build/migrate-to-built-in-kotlin
> 
> Android Gradle plugin 9.0 introduces built-in Kotlin support and enables it by default. That means you no longer have to apply the org.jetbrains.kotlin.android (or kotlin-android) plugin in your build files to compile Kotlin source files. With built-in Kotlin, your build files are simpler and you can avoid compatibility issues between AGP and the kotlin-android plugin.

Everything builds normally for me.

File: dependencies.gradle (hunk 6 above)
Comment (DrakslerT) [reply dependencies.gradle]: Updated to 9.3

File: processor/build.gradle (hunk 3 above)
Comment (DrakslerT) [reply processor/build.gradle:43]: I have now tested it with a local deploy of the artifacts and running the sample app has no issues. 
It seems it is okay to me, but I am not sure how to test this more thoroughly

Comment (KCeh) [reply processor/build.gradle:43]: Ideally, we would build agains other project that uses this lib, just as a sanity check

File: build.gradle (hunk 5 above)
Comment (KCeh) [reply build.gradle:15]: Ok but I think we need to do some migration sooner or later; the legacy dependency name doesn't sound very promising

**Review by KCeh — APPROVED** [review]
